File: frontend/lib/api/analytics.ts
import { apiClient } from './client';

export interface RecentInterview {
  id: number;
  title: string;
  status: string;
  user_id: number;
  created_at: string;
  completed_at: string | null;
  overall_score: number | null;
}

export interface OverviewData {
  total_interviews: number;
  completed_interviews: number;
  active_interviews: number;
  average_score: number | null;
  average_duration_minutes: number | null;
  total_code_runs: number;
  total_ai_messages: number;
  recent_interviews: RecentInterview[];
  recent_session_events: SessionEvent[];
}

export interface SessionEvent {
  id?: number;
  interview_id: number;
  event_type: string;
  created_at: string;
}

export interface InterviewAnalytics {
  interview_id: number;
  title: string;
  status: string;
  language: string;
  user_id: number;
  user_email: string | null;
  problem_id: number | null;
  problem_title: string | null;
  code_submissions_count: number;
  latest_code_result: object | null;
  ai_evaluation: {
    overall_score: number | null;
    technical_score: number | null;
    code_quality_score: number | null;
    communication_score: number | null;
    problem_solving_score: number | null;
    feedback_summary: string | null;
  } | null;
  session_events: SessionEvent[];
  message_count: number;
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
}

export interface CandidateAnalytics {
  user_id: number;
  email: string;
  full_name: string | null;
  total_interviews: number;
  completed_interviews: number;
  average_score: number | null;
  best_score: number | null;
  latest_feedback: string | null;
  total_code_runs: number;
  total_messages: number;
}

export interface SystemActivity {
  recent_session_events: SessionEvent[];
  recent_code_submissions: object[];
  recent_completed_interviews: object[];
}

export const analyticsApi = {
  getOverview: () => apiClient.get<OverviewData>('/api/v1/analytics/overview'),
  getInterviewAnalytics: (id: number) => apiClient.get<InterviewAnalytics>(`/api/v1/analytics/interviews/${id}`),
  getCandidateAnalytics: (id: number) => apiClient.get<CandidateAnalytics>(`/api/v1/analytics/candidates/${id}`),
  getSystemActivity: () => apiClient.get<SystemActivity>('/api/v1/analytics/system-activity'),
};

File: src/schemas/analytics.py
"""Analytics Pydantic schemas."""

from typing import Optional, List, Any
from pydantic import BaseModel


class RecentInterview(BaseModel):
    id: int
    title: str
    status: str
    user_id: int
    created_at: str
    completed_at: Optional[str] = None
    overall_score: Optional[int] = None


class OverviewResponse(BaseModel):
    total_interviews: int
    completed_interviews: int
    active_interviews: int
    average_score: Optional[float] = None
    average_duration_minutes: Optional[float] = None
    total_code_runs: int
    total_ai_messages: int
    recent_interviews: List[RecentInterview] = []
    recent_session_events: List[dict] = []


class InterviewAnalyticsResponse(BaseModel):
    interview_id: int
    title: str
    status: str
    language: str
    user_id: int
    user_email: Optional[str] = None
    problem_id: Optional[int] = None
    problem_title: Optional[str] = None
    code_submissions_count: int
    latest_code_result: Optional[dict] = None
    ai_evaluation: Optional[dict] = None
    session_events: List[dict] = []
    message_count: int
    started_at: Optional[str] = None
    completed_at: Optional[str] = None
    created_at: str


class CandidateAnalyticsResponse(BaseModel):
    user_id: int
    email: str
    full_name: Optional[str] = None
    total_interviews: int
    completed_interviews: int
    average_score: Optional[float] = None
    best_score: Optional[int] = None
    latest_feedback: Optional[str] = None
    total_code_runs: int
    total_messages: int


class SystemActivityResponse(BaseModel):
    recent_session_events: List[dict] = []
    recent_code_submissions: List[dict] = []
    recent_completed_interviews: List[dict] = []
